connect/runtime/src/main/java/org/apache/kafka/connect/runtime/AbstractHerder.java

```diff
@@ -230,16 +231,17 @@ public static ConfigInfos generateResult(String connType, Map<String, ConfigKey>
             configValueMap.put(configName, configValue);
             if (!configKeys.containsKey(configName)) {
                 configValue.addErrorMessage("Configuration is not defined: " + configName);
-                configInfoList.add(new ConfigInfo(null, convertConfigValue(configValue)));
+                configInfoList.add(new ConfigInfo(null, convertConfigValue(configValue, null)));
             }
         }
 
         for (String configName: configKeys.keySet()) {
             ConfigKeyInfo configKeyInfo = convertConfigKey(configKeys.get(configName));
+            Type type = configKeys.get(configName).type;
```

Comment:
Like I said, not a big deal and I'll commit as is. But by using entrySet, I didn't mean not to use key at all, just that something like:

```
for (Map.Entry<String, String> entry : configKeys.entrySet()) {
    String configName = entry.getKey();
    Type type = entry.getValue();
```

would avoid doing the extra lookup and would still let you use the key directly as needed.